api/src/main/java/ai/djl/ndarray/index/full/NDIndexFullPick.java

```diff
@@ -72,6 +73,7 @@ public static Optional<NDIndexFullPick> fromIndex(NDIndex index, Shape target) {
                             "Only rank-1 indexing array is supported for pick");
                 }
                 fullPick = new NDIndexFullPick(indexElem, axis);
+                fullPick.setIndexTake(true);
```

Comment:
Here, I tried to use take for MXNet engine too see https://github.com/deepjavalibrary/djl/pull/1802, and the PR tests all passed. But it caused the https://github.com/deepjavalibrary/djl/issues/1800. Here in your edition, setting fullPick.setIndexTake(true); will trigger this problem again.

Reply:
`NDArray.take(NDArray)` returns a shape that is not useful for the functionality of this pull request:

`Returns a partial NDArray pointed by index according to linear indexing, and the output is of the same shape as index.`

Given `ndArray` of shape e.g. `(a,b,c,d)` and `index` of shape `(e) `, then I assumed initially that `ndArray.take(index)` would return an array of shape` (e,b,c,d)` .  Therefore I restyled my initial implementation, using NDIndexFullTake .  If anybody has a better idea, let me know, at least the test cases go green now.